### DISCUSSION TITLE
Passing in a paremeter to EvaluateFunctionAsync

### DISCUSSION THREAD

**Discussion by Naelsh** [hardkoded/puppeteer-sharp#2844] [Q&A] [answered]
I'm currently trying to get specific children from an element but can't seem to get it to work.

```cs
var details = await page.EvaluateFunctionAsync<Object>(@"
      (index) => {
          const block = document.querySelector('div.grupo_prefiles:nth-child(index)');
          return block.querySelector('#at1_0') ? block.querySelector('#at1_0 > .valor').innerText : 'No Data';
      }", childIndex);
```

Error that is thrown:
Failed to execute 'querySelector' on 'Document': 'div.grupo_prefiles:nth-child(index)' is not a valid selector.

I've tried it with  a hardcoded value instead such as 3 instead of index and that works just fine.

I'm sure that it is something I've missed, so if anyone has a clue I would be so so happy!

**Comment by kblok** [comment, marked as answer]
You have `index` inside the string. It should be something like this
```
const block = document.querySelector(`div.grupo_prefiles:nth-child(${index})`);
```

**Reply by Naelsh** [reply]
I stumble on the same problem with that :/

SyntaxError: Failed to execute 'querySelector' on 'Document': 'div.grupo_prefiles:nth-child(${index})' is not a valid selector.

**Reply by kblok** [reply]
Did you use backticks? It is not a puppeteer issue. Your CSS selector is wrong.

**Reply by Naelsh** [reply]
Oh wow! thank you, that was the problem :) got it working now!
